OverlayPlugin/SanityChecker.cs

```diff
@@ -10,6 +10,8 @@ namespace RainbowMage.OverlayPlugin
 {
     public static class SanityChecker
     {
+        private static Dictionary<string, Assembly> LoadedAssemblies = new Dictionary<string, Assembly>();
```

Comment:
Isn't this just a filtered version of `AssemblyResolver.assemblyCache`? The only difference is that this dictionary only contains OverlayPlugin assemblies while the `assemblyCache` contains all assemblies loaded from OverlayPlugin's `libs` folder.